### ISSUE TITLE
Demo of a Godot game completely crashes computer upon start up.

### ISSUE THREAD

**Issue by TheNetherPug** [godotengine/godot#95437]
### Tested versions

Godot 3.5.3

### System information

Windows 11 - Godot 3.5.3 - Nvidia GTX 1650 - Intel i5 9300H CPU - 8GB of RAM 

### Issue description

When playing the Index Purger Demo on steam, my computer shuts itself down entirely.
https://store.steampowered.com/app/2742480/Index_Purger/

I would like to point out that this appears to be an **engine issue**, and the developer behind Index Purger (Dadaskis) has pointed me to create a github issue in here. Neither of us have been able to find a solution to this issue.

Upon starting Index Purger via steam or via the executable in the game files, the game loads up and then suddenly crashes the computer I am using.

I am using an Acer Nitro 5 laptop. The graphics drivers are up to date, I have edited my TdrDelay to be 60 seconds and I have run `sfc /scannow`. None of which solved the problem.

Attached below are all the log files from each crash during our testing.

[godot.log](https://github.com/user-attachments/files/16590780/godot.log)
[godot2024-08-12T12.43.19.log](https://github.com/user-attachments/files/16590781/godot2024-08-12T12.43.19.log)
[godot2024-08-12T13.05.07.log](https://github.com/user-attachments/files/16590782/godot2024-08-12T13.05.07.log)
[godot2024-08-12T13.26.25.log](https://github.com/user-attachments/files/16590783/godot2024-08-12T13.26.25.log)
[godot2024-08-12T13.45.48.log](https://github.com/user-attachments/files/16590785/godot2024-08-12T13.45.48.log)
[godot2024-08-12T14.06.15.log](https://github.com/user-attachments/files/16590786/godot2024-08-12T14.06.15.log)
[godot2024-08-12T14.14.18.log](https://github.com/user-attachments/files/16590787/godot2024-08-12T14.14.18.log)
[godot2024-08-12T14.31.25.log](https://github.com/user-attachments/files/16590788/godot2024-08-12T14.31.25.log)

### Steps to reproduce

1. Download the Index Purger Demo from steam
2. Launch it
3. If the issue occurs, your machine will shut itself down immediately

### Minimal reproduction project (MRP)

There is no MRP for this issue.

**Comment by Dadaskis** [comment]
I would like to add that crash occurs right at the beginning of starting loading screen. All of the previous actions (updating of Nvidia drivers, TDR Delay change, usage of async/sync compilation, `sfc /scannow`) helped to delay the crash for a little bit. Before it didn't go into fullscreen and didn't play music, now it both going fullscreen and playing music. I'm out of ideas, because his hardware is more than sufficient to run this game, not to mention running any GLES3 applications

**Comment by Calinou** [comment]
Can you try running the game with the `--video-driver GLES2` [Steam launch option](https://help.steampowered.com/en/faqs/view/7D01-D2DD-D75E-2955)? Also try the `--no-window` launch option.

**Comment by TheNetherPug** [comment]
> Can you try running the game with the `--video-driver GLES2` [Steam launch option](https://help.steampowered.com/en/faqs/view/7D01-D2DD-D75E-2955)? Also try the `--no-window` launch option (no window will appear, but the game might not crash anymore, helping narrow down the issue).

Thank you so much for the response

I launched with `--video-driver GLES2` via terminal, which allows me to _almost_ see the entire loading animation, but then it crashes right before the end of it. Launching with the `--no-window` via terminal did not do anything and launched the game with the window

**Comment by Dadaskis** [comment]
We have decided to try installing older GPU drivers (March) and this resulted in even worse situation. Now the game crashes even faster than before. So it may be related to rendering now for sure

**Comment by akien-mga** [comment]
This CPU and GPU should have no problem running Godot 3, so this suggests a further system specific issue, otherwise we'd be swarmed with duplicate reports.

So I would suggest trying to identify what is installed on that computer that may differ from other typical setups.

Driver version was a good first try, then I'd check whether you're using things like ReShade or other graphics/games oriented layers which inject themselves in applications. Trying to disable the Steam overlay may help too.

**Comment by TheNetherPug** [comment]
> This CPU and GPU should have no problem running Godot 3, so this suggests a further system specific issue, otherwise we'd be swarmed with duplicate reports.
> 
> So I would suggest trying to identify what is installed on that computer that may differ from other typical setups.
> 
> Driver version was a good first try, then I'd check whether you're using things like ReShade or other graphics/games oriented layers which inject themselves in applications. Trying to disable the Steam overlay may help too.
> 
> Another option could be faulty RAM or SSD but that's a long shot.

Heyo, thank you for the response!

I do not have anything like ReShade installed on this machine, and the game also crashes even when launched from the executable directly without steam opened.

Another thing to note: I have tested an older version of this game to see if it may have been an update to the game, and the issue still persists. The older version is stored on the SSD and the Steam version is stored on the HDD, so I don't think it is an SSD/HDD issue. Other Godot 3.X games work just fine, as well as the standalone engine itself. Additionally, Index Purger also crashes when launched from the editor.

**Comment by Dadaskis** [comment]
We've went back to the very first export of the game and we tried to make a minimal reproduction of it, which looks like that:
![image](https://github.com/user-attachments/assets/15bd88a5-4cf9-456b-94fb-619bfc9b57ab)

I have a suspicion that my custom-written classes named `SpatialUbershader` and `UserSpatialUbershader` may cause the issue. Would you please take a quick look on them? In this version they compile the shaders all the time, while modern Purger actually has a cache of them.

[effect_config.gd.txt](https://github.com/user-attachments/files/16603595/effect_config.gd.txt)
[spatial_ubershader.gd.txt](https://github.com/user-attachments/files/16603596/spatial_ubershader.gd.txt)
[user_spatial_ubershader.gd.txt](https://github.com/user-attachments/files/16603597/user_spatial_ubershader.gd.txt)

I want to note that `UserSpatialUbershader` has a lot of repetitive code that is generated in editor

**Comment by Dadaskis** [comment]
Today is the best worst day of my life... 2 crippling bugs got fixed... Yeah i'm writing it before TNP's message

**Comment by TheNetherPug** [comment]
The problem has been resolved!

The battery in my laptop was improperly reconnected since the last time I had opened it. My laptop is normally always plugged in which prevented me from noticing this issue. I don't know what happened in Godot my laptop to use the battery when playing Index Purger, but since it was improperly connected, it caused a full shut down of the laptop.

I reconnected the battery properly and now the issue is fixed. Thank you all for all of your assistance in trying to solve this very stupid issue, lol
